sylvia-derive/src/querier.rs

```diff
@@ -0,0 +1,119 @@
+use convert_case::{Case, Casing};
+use proc_macro2::{Ident, TokenStream};
+use proc_macro_error::emit_error;
+use quote::quote;
+use syn::parse::{Parse, Parser};
+use syn::spanned::Spanned;
+use syn::{GenericParam, ImplItem, ItemImpl, ItemTrait, TraitItem};
+
+use crate::check_generics::CheckGenerics;
+use crate::crate_module;
+use crate::message::{MsgField, MsgVariant};
+use crate::parser::{MsgAttr, MsgType};
+
+pub struct Querier<'a> {
+    variants: Vec<MsgVariant<'a>>,
+}
+
+impl<'a> Querier<'a> {
```

Comment:
In general - I am pretty sure almost the whole of this thing belongs to the message variant, which is built first from the `ItemTrait` or `ItemImpl`, and then just some `emit_querier` is called on it (more or less, probably something more on that which comes out while implementing).